Issue6335: RecordInlineCell tree refactor with RecordInlineCellContext

hansol-y opened the pull request:
Fixes #6335 

This pull request is for issue #6335: Refactor RecordInlineCell tree with a Context to avoid props drilling. For the refactoring, this PR made changes as below:
- Created new script RecordInlineCellContext.tsx: Defining a context to pass in useContext()
- Updated `RecordInlineCell.tsx`: Passing the necessary props as context values, wrapping with RecordInlineCellContext.Provider
- Updated `RecordInlineCellContainer.tsx`: Passing the props to RecordInlineContainer as RecordInlineCellContext

Using RecordInlineCellContext, I believe the context goes to the top of the hierarchy and passed to the required layers without going through several layers. However, please let me know if I understood the issue incorrectly or it is not solved properly.

Thank you in advance for your review!

lucasbordeau commented on packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx:
```diff
@@ -146,20 +154,18 @@ export const RecordInlineCellContainer = ({
-            showLabel,
-            isCentered,
-          }}
+          displayModeContent={displayModeContent}
```
Nice, could you try avoid drilling down in RecordInlineCellValue and call those props where they are need at the lowest level possible ? 

We should avoid calling RecordInlineCellContext in the FieldDisplay and FieldInput hierarchy, as they are called both by RecordInlineCell and RecordTableCell

hansol-y replied:
Thank you for the feedback! I updated the scripts to avoid drilling down into RecordInlineCellValue in RecordInlineCellContainer and getting the context directly from RecordInlineCellValue. Please let me know any further changes are required!

lucasbordeau reviewed (commented):
- You should test if the context is defined directly in `useRecordInlineCellContext` hook and throw otherwise, because now it breaks the rules of hook.

It's not that for props drilling, we can still refactor way more.

I'll give you an example : 

`RecordInlineCellButton` Only needs buttonIcon, so it could be extracted from the context only in this last child component.
`RecordInlineCellDisplayMode` only needs showLabel, the rest can be extracted directly inside RecordInlineCellDisplayMode from context

It's almost done

hansol-y commented:
Hi @lucasbordeau,

Thank you for your feedback. Based on your suggestions, I made the following changes:

**RecordInlineCellDisplayMode**: Now retrieves values from `useRecordInlineCellContext` instead of directly assigning them.
**RecordInlineCellValue**: Removed values passed through `<RecordInlineCellDisplayMode>` as they are now retrieved through `useRecordInlineCellContext` + Removed the null check for `RecordInlineCellContextProps`.
**RecordInlineCellContainer**: Removed the null check for `RecordInlineCellContextProps`.
**RecordInlineCellContext**: Defined default `RecordInlineCellContextProps` to avoid null values.
**ActivityTargetsInlineCell**: Now passes values using `RecordInlineCellContext.Provider` instead of `RecordInlineCellContainer`.

I also reviewed `RecordInlineCellButton` and other related scripts but did not find any properties defined in `RecordInlineCellContext`, so I left them unchanged.

Please let me know if any further adjustments are needed. Thank you again for your detailed feedback and support. It has been incredibly helpful in my learning process.

lucasbordeau commented:
@hansol-y Please could you create a branch in your fork, because it's hard for me to review if it's on branch main of a fork.

hansol-y commented:
@lucasbordeau  Please check this [branch](https://github.com/hansol-y/twenty/tree/issue-6335-prevent-prop-drilling) and feel free to let me know if I can do any other actions for you!

lucasbordeau commented:
@hansol-y I meant a branch + another PR or change this PR's branch to merge into main.

hansol-y commented:
Hey @lucasbordeau , I created new [PR](https://github.com/twentyhq/twenty/pull/6537). Sorry for my late action! I missed your latest comment😭 And please feel free to let me know if any other changes are required. I'm closing this pr.
